Modules/_testcapimodule.c

```diff
@@ -3420,6 +3420,172 @@ test_pyobject_setallocators(PyObject *self)
     return test_setallocators(PYMEM_DOMAIN_OBJ);
 }
 
+/* Part of the following code is inherited from the pyfailmalloc project
+ * written by Victor Stinner. */
+static struct {
+    int installed;
+    PyMemAllocatorEx raw;
+    PyMemAllocatorEx mem;
+    PyMemAllocatorEx obj;
+} FmHook;
+
+typedef struct fm_filter_t {
+    void *data;
+    int (*has_memory) (struct fm_filter_t *);
+} fm_filter_t;
```

Comment:
This structure seems very generic, whereas the implementation is very specific.

Do we really need a configurable callback? Why not always call the same filter function and just store somewhere counters?